portal/code/tractus-x-portal/src/helpers/UserService.js

```diff
@@ -60,7 +60,11 @@ const getDomain = () => realm;//_kc.tokenParsed?.split('/').pop();
 
 const getCompany = () => _kc.tokenParsed?.email.split('@')[1].split('.')[0].toUpperCase();
```

Comment:
shared idp users will have two claims: 'tenant' and 'organization' that are mapped from per-idp mapped attributes. 'tenant' refers to the idp id as it is defined in keycloak ('idp1234'), 'organization' refers to the company-name that is being used in invitation and being used as the realms displayname (that is what gets displayed on the per-idp selection-buttons resp. the per realm login-screen).
This information is more reliable as parsing the email-address (a user might use e.g. googlemail.com ;-) )

Reply:
agree - this is a q'n'd line of code from the Speedboat, where we quickly needed something to show a company name

Reply:
didn't receive the attribute 'organization' in the token. so we are now temporarily using the first part of the tenant ID as company name until a proper mapping exists.